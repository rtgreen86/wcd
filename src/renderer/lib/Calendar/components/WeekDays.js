import React from 'react';
import useDaysOfWeek from '../hooks/useDaysOfWeek';

export default function WeekDays() {
  const daysOfWeek = useDaysOfWeek();
  return <tr>{daysOfWeek.map(({ cellNumber, marks, caption }) => <td key={ cellNumber } className={ marks.join(' ') }>{ caption }</td>)}</tr>;
}
