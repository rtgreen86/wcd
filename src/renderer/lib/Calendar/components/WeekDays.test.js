import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import WeekDays from './WeekDays.jsx';

describe('WeekDays', () => {
  describe('rendered', () => {
    test('match snapshot', () => {
      const actual = render(<table><thead><WeekDays /></thead></table>);
      expect(actual.asFragment()).toMatchSnapshot();
    });
  });
});
