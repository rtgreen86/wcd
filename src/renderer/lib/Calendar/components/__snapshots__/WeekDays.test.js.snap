// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`WeekDays rendered match snapshot 1`] = `
<DocumentFragment>
  <table>
    <thead>
      <tr>
        <td
          class="weekend"
        >
          Sun
        </td>
        <td
          class=""
        >
          Mon
        </td>
        <td
          class=""
        >
          Tue
        </td>
        <td
          class=""
        >
          Wed
        </td>
        <td
          class=""
        >
          Thu
        </td>
        <td
          class=""
        >
          Fri
        </td>
        <td
          class="weekend"
        >
          Sat
        </td>
      </tr>
    </thead>
  </table>
</DocumentFragment>
`;
